src/quest_manager/models.py

```diff
@@ -975,3 +989,150 @@ def get_minutes_to_complete(self):
+
+    This model is used for all types of questions, including multiple choice, short answer, file upload, etc.
+    """
+
+    instructions = models.TextField(
+        help_text=(
+            "Appears to the student when they are completing the quest. Should provide "
+            "specific guidance, a prompt, or a problem statement for the student to address."
+        )
+    )
+    solution_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, allowing for an easy comparison of "
+            "the student's submission to an ideal or example solution."
+        ),
+    )
+    solution_file = RestrictedFileField(
+        blank=True,
+        null=True,
+        upload_to="quest/question/solution/%Y/%m/%d",
+        help_text=(
+            "This file will appear to the marker, allowing them to easily compare "
+            "the student's submission to an ideal or example solution for file-based questions."
+        ),
+    )
+
+    # Each tuple contains the file type as both the stored value and the display value
+    # e.g. ("media", "Media") or ("image", "Image")
+    allowed_file_type = models.CharField(
+        max_length=50,
+        choices=[(key, key.capitalize()) for key in FILE_MIME_TYPES.keys()],
+        default="media",
+        help_text=(
+            "The types of files that can be uploaded by a student for this question."
+        ),
+    )
+
+    marker_notes = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, providing additional "
+            "context and info, if needed."
+        ),
+    )
+    required = models.BooleanField(
+        default=True,
+        help_text=(
+            "If this field is unchecked, the student can complete the quest without "
+            "answering this question."
+        ),
+    )
+    datetime_created = models.DateTimeField(auto_now_add=True)
+    datetime_last_edit = models.DateTimeField(auto_now=True)
+    ordinal = models.PositiveIntegerField(
+        default=1,
+        help_text=(
+            "Questions will be displayed to the student in order from lowest ordinal to "
+            "highest ordinal."
+        ),
+    )
+
+    quest = models.ForeignKey(Quest, on_delete=models.CASCADE)
+    type = models.CharField(
+        max_length=20, choices=QuestionType.choices, default=QuestionType.SHORT_ANSWER
+    )
+
+    class Meta:
+        ordering = ["ordinal"]
+        # ensure two questions in the same quest cannot have the same ordinal
+        unique_together = ["quest", "ordinal"]
+
+    def __str__(self):
+        """Return string representation in format of: [Quest Name] ([Question Type] - [Question Ordinal])
+        e.g. "Quest 1 (short_answer - 1)" for the first question in the quest with name "Quest 1"
+        """
+        return f"{self.quest.name} ({self.type} - {self.ordinal})"
```

Comment:
I don't think the quest name here is necessary.  When would a question be viewed independent of it's quest? The quest context will always be there, (i.e. the question will always be listed with the quest or with a submission so I think it just needs "ordinal - type".  In the detail view we can add the quest name explicitly.

If you think the quest name is still necessary, it should be the part in brackets, not the ordinal and type.

Reply:
```py
    def __str__(self):
        """Return string representation in format of: [Question Ordinal] - [Question Type]
        e.g. "1 - short_answer" for the first question in the quest with type short_answer
        """
        return f"{self.ordinal} - {self.type}"
```